sdk/python/kfp/v2/components/experimental/component_spec.py

```diff
@@ -14,10 +14,18 @@
 """Definitions for component spec."""
 
 import dataclasses
-import enum
 import itertools
 import json
 from typing import Any, Dict, Mapping, Optional, Sequence, Union
+try:
+    from typing import OrderedDict
+except ImportError:
+    from collections import OrderedDict
+
+try:
+    from typing import ForwardRef
+except ImportError:
+    from typing import _ForwardRef as ForwardRef
```

Comment:
You can use `from pydantic.typing import ForwardRef`. That will probably solve your Python3.6 issues.

Reply:
Seems like it leads to another error. Tried and realize the pure `" "` works better than `ForwardRef(" ")`